### PR TITLE
AWS transport selector with tests

### PR DESCRIPTION
**Pull request by danotorrey**
**Please review and merge #116 first. Any merge conflicts on this PR will go away once #116 is merged.**

This adds the logic, which chooses the correct transport for a particular AWSMessageType.

The actual KinesisTransport will plug directly into this structure once ported over.

### CODE REVIEW

**Comment by ceruleancee** [comment]
conflicting files need to be resolved

File: src/main/java/org/graylog/integrations/aws/transports/KinesisTransport.java [hunk 1]
```diff
@@ -61,7 +62,7 @@ public void doLaunch(MessageInput input) throws MisfireException {
 
     @Override
     public void doStop() {
-
+        LOG.info("Stopping");
```
Comment (ceruleancee) [inline src/main/java/org/graylog/integrations/aws/transports/KinesisTransport.java]: LOG.info("Stopping Kinesis Transport.");

**Comment by danotorrey** [comment]
@ceruleancee The merge conflicts and log entries have been fixed. Ready for another review.

**Comment by danotorrey** [comment]
@ceruleancee The log entries have been fixed. Ready for another review.

The merge conflicts have been fixed, but may still show on this PR until #116 is merged.

File: src/test/java/org/graylog/integrations/AWSTestingUtils.java [hunk 2]
```diff
@@ -0,0 +1,61 @@
+package org.graylog.integrations;
```
[the remark this answers is not in the document]
Comment (danotorrey) [reply src/test/java/org/graylog/integrations/AWSTestingUtils.java]: Thanks. The intention is to keep `src/test/java/org/graylog/integrations/aws/AWSTestingUtils.java`. Fixed in https://github.com/Graylog2/graylog-plugin-integrations/pull/117/commits/65471a2d4189f43233ed48a659f1f52ad4d4e45b.

**Pull request merged**